scipy/optimize/_linprog_util.py

```diff
@@ -1143,12 +1150,11 @@ def vstack(blocks):
     l_free = np.logical_and(lb_none, ub_none)
     i_free = np.nonzero(l_free)[0]
     n_free = len(i_free)
-    A1 = hstack((A1, zeros((A1.shape[0], n_free))))
     c = np.concatenate((c, np.zeros(n_free)))
     if x0 is not None:
         x0 = np.concatenate((x0, np.zeros(n_free)))
-    A1[:, range(n_ub, A1.shape[1])] = -A1[:, i_free]
-    c[np.arange(n_ub, A1.shape[1])] = -c[i_free]
+    A1 = hstack((A1[:, :n_ub], -A1[:, i_free]))
```

Comment:
Stack `A1[:, i_free]` directly rather than stacking zeros and filling in the values. Makes sense. 👍